File: books-practice/grokking-algorithms/binary-search.py
def binary_search(list, item):
  low=0  #0
  high=len(list)-1 #4 -> 0,1,2,3,4
  
  while low <= high: # 0 <=4
    print('run')
    mid=(low+high) // 2 # 2
    guess=list[mid] # mid position -> 5
    if guess == item: # 9 == 4 X
      return mid 
    if guess > item: # 9 > 7
      high=mid-1 # 3
    else:
      low=mid+1
  return None

mi_List = [1,3,5,7,9]
print(binary_search(mi_List, 7))
print(binary_search(mi_List,-1))

File: books-practice/grokking-algorithms/binary-search.ts
function binary_search(list: Array<number>, element: number) {
  let low = 0;
  let high = list.length - 1;

  while (low <= high) {
    let mid = Math.floor((low + high) / 2);
    let guess = list[mid];
    if (guess === element) {
      return mid;
    }
    if (guess > element) {
      high = mid - 1;
    } else {
      low = mid + 1;
    }
  }
  return null;
}
function generarNumeros() {
  let numeros: Array<number> = [];
  for (let i = 0; i <= 20; i++) {
    numeros.push(i);
  }
  return numeros;
}

const list = generarNumeros()
console.log(list.length)
console.log(binary_search(list, 12))